packages/core/src/common/keybinding.ts

```diff
@@ -150,16 +154,25 @@ export class KeybindingRegistry {
      */
     registerKeybinding(binding: Keybinding) {
         try {
-            const existingBindings = this.getKeybindingsForKeyCode(KeyCode.parse(binding.keybinding));
-            if (existingBindings.length > 0) {
-                const collided = existingBindings.filter(b => b.context === binding.context);
+            const existingBindings = this.getKeybindingsForKeySequence(KeySequence.parse(binding.keybinding));
+            if (existingBindings.full.length > 0) {
+                const collided = existingBindings.full.filter(b => b.context === binding.context);
                 if (collided.length > 0) {
                     this.logger.warn('Collided keybinding is ignored; ',
                         Keybinding.stringify(binding), ' collided with ',
                         collided.map(b => Keybinding.stringify(b)).join(', '));
                     return;
                 }
             }
+            if (existingBindings.partial.length > 0) {
+                const shadows = existingBindings.partial.filter(b => b.context === binding.context);
+                if (shadows.length > 0) {
+                    this.logger.warn('Shadowing keybinding is ignored; ',
```

Comment:
I thought shadowing keybinding means that a keybinding from one scope shadows a keybinding from another. I think here 2 keybindings are colliding again or overlapping. Do I miss something?

Reply:
It depends on your definition of shadowing... I've used the term to describe that if you had `ctrl-x ctrl-t` and try to register `ctrl-x` then it would shadow `ctrl-x ctrl-t` since you could not get to that keybinding.  I could use the word colliding too but that to me meant more that it was using the same exact keybinding...

Reply:
@akosyakov you will get a warning in that case too since if it will check do I have a keybinding for `ctrl-x ctrl-t` and it will find a partial keybinding for it in `ctrl-x` and refuse to register it. In this case `ctrl-x` would shadow `ctrl-x ctrl-t` ... but I think I have a problem if you have `ctrl-x ctrl-t` and try to regiter `ctrl-x ctrl-a` you'll get in the same conditions :( I have to fix that...